indico/web/forms/validators.py

```diff
@@ -209,3 +209,17 @@ def _condition(form, field):
             return form._fields.get(field_name).data
 
         super(UsedIfChecked, self).__init__(_condition)
+
+
+class MaxDuration(object):
+    """Validates if TimeDeltaField value doesn't exceed `max_duration`"""
+
+    def __init__(self, max_duration=None):
+        self.max_duration = max_duration
+
+    def __call__(self, form, field):
+        if field.data is None or self.max_duration is None:
+            return
+        if field.data > self.max_duration:
+            raise ValidationError(_('Value for the {} can not exceed {}'
```

Comment:
not a good error message, it'll be very strange once you start translating things. I'd get rid of the label and just use `"Duration may not exceed ..."`